### ISSUE TITLE
exclude_dir is broken in 6.1.8

### ISSUE THREAD

**Issue by jefflee-icesat2** [Fortran-FOSS-Programmers/ford#391]
(Python 3.9) This has to do with mixing strings and PosixPath instances.

This (ugly) fixes  it in fortran_project.py, but am unsure if there are unintended consequences later on...

```
import pathlib
...
    def recursive_dir_list(self, topdir, skip):
        dir_list = []
        for entry in os.listdir(topdir):
            abs_entry = pathlib.Path(os.path.join(topdir, entry)).absolute()
```

**Comment by ZedThree** [comment]
Should be fixed in #394.

@ajdawson @jefflee-icesat2 If you could give that branch a whirl on your codes, and let me know if it works for you, that would be great! I've added test cases, but it would be good to know if I've let anything slip through the cracks

**Comment by ajdawson** [comment]
This branch appears to work in our docs build, thanks @ZedThree.
